cucumber/authenticators_jwt/features/authn_jwt_validate_restrictions.feature

```diff
@@ -0,0 +1,245 @@
+Feature: JWT Authenticator - Validate restrictions
+  Background:
+    Given I initialize JWKS endpoint with file "myJWKs.json"
+    And I load a policy:
+    """
+    - !policy
+      id: conjur/authn-jwt/raw
+      body:
+      - !webservice
+        annotations:
+          description: Authentication service for JWT tokens, based on raw JWKs.
+
+      - !variable
+        id: jwks-uri
+
+      - !variable
+        id: token-app-property
+
+      - !group users
+
+      - !permit
+        role: !group users
+        privilege: [ read, authenticate ]
+        resource: !webservice
+    """
+    And I am the super-user
+    And I successfully set authn-jwt jwks-uri variable with value of "myJWKs.json" endpoint
+
+  Scenario: General annotation invalid service specific annotation, 200 OK
+    Given I have a "variable" resource called "test-variable"
+    And I extend the policy with:
+    """
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/project-id: myproject
+        authn-jwt/aud: myaud
+        authn-jwt/raw/project-id: myproject
+        authn-jwt/raw/aud: myaud
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/users
+      member: !host myapp
+    """
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    And I add the secret value "test-secret" to the resource "cucumber:variable:test-variable"
+    And I permit host "myapp" to "execute" it
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "project-id": "myproject",
+      "aud": "myaud"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then host "myapp" has been authorized by Conjur
+    And I successfully GET "/secrets/cucumber/variable/test-variable" with authorized user
+    And The following appears in the log after my savepoint:
+    """
+    cucumber:host:myapp successfully authenticated with authenticator authn-jwt service cucumber:webservice:conjur/authn-jwt/raw
+    """
+
+  Scenario: General annotation and without service specific annotations, 401 Error
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    Given I extend the policy with:
+    """
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/project-id: myproject
+        authn-jwt/aud: myaud
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/users
+      member: !host myapp
+    """
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "project-id": "myproject",
+      "aud": "myaud"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00099E Role must have at least one relevant annotation
+    """
+
+  Scenario: General annotation and invalid service specific annotation, 401 Error
```

Comment:
What is the different with `General annotation invalid service specific annotation, 200 OK`

Reply:
If there is annotation with valid service but invalid value we want to fail authentication
If there is annotation with invalid service we ignore it no matter what is the value. 

I will change to better namings